### PR TITLE
Add TREC Covid Reranking Task

### PR DESCRIPTION
**Pull request by justinborromeo**
Command:

```
python -um pygaggle.run.evaluate_trec_covid_ranker                  
            --method t5                                                
            --model castorini/monot5-base-med-msmarco
            --dataset data/trec_covid 
            --model-type t5-base
            --task trec-covid 
            --index-dir indexes/lucene-index-cord19-abstract-2020-07-16
            --batch-size 16
            --output-file runs/run.monot5.trec_covid.tsv
```

Qrels file: qrels-covid_d5_j4.5-5.txt
Input Runfile: expanded.anserini.final-r5fusion1.txt
Queries: topics-covid-round5.xml

trec_eval output: https://gist.github.com/justinborromeo/cb6cff577c926d47a78ba9170add5dd9

The scores shown in the trec_eval output seem a little bit low.  I'll add documentation after this has been reviewed.

### CODE REVIEW

File: pygaggle/data/msmarco.py [hunk 1]
```diff
@@ -61,7 +67,7 @@ def load_queries(cls,
                      run) -> List[MsMarcoExample]:
         queries = []
         with open(path) as f:
-            for i, line in enumerate(f):
+            for _, line in enumerate(f):
```
Comment (rodrigonogueira4) [inline pygaggle/data/msmarco.py:70]: remove enumerate if we don't use it?

File: pygaggle/rerank/transformer.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
                           PreTrainedTokenizer,
                           T5ForConditionalGeneration)
 import torch
-
+import logging
```
Comment (ronakice) [inline pygaggle/rerank/transformer.py:8]: unused?

File: pygaggle/data/segmentation.py [hunk 3]
```diff
@@ -32,17 +32,24 @@ def __init__(self, max_characters=10000000):
...
+                    if not document.title == '':
+                        segment_text = document.title + '. ' + segment_text
+                    segmented_doc.append(Text(segment_text, dict(docid=document.metadata["docid"])))
+                    if i + seg_size >= len(sentences):
+                        end_idx += i/stride + 1
+                        doc_end_indexes.append(int(end_idx))
+                        break
         return SegmentGroup(segmented_doc, doc_end_indexes)
 
     def aggregate(self, documents: List[Text], segments_group: SegmentGroup, method: str = "max") -> List[Text]:
```
Comment (rodrigonogueira4) [inline pygaggle/data/segmentation.py:55]: Add a short docstring to each method and one for the file describing what it does in general?

File: pygaggle/run/evaluate_trec_covid_ranker.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    return SequenceClassificationTransformerReranker(model, tokenizer)
+
+
+def construct_bm25(options: DocumentRankingEvaluationOptions) -> Reranker:
+    return Bm25Reranker(index_path=str(options.index_dir))
+
+
+def main():
+    apb = ArgumentParserBuilder()
+    apb.add_opts(opt('--task',
```
Comment (rodrigonogueira4) [inline pygaggle/run/evaluate_trec_covid_ranker.py:117]: Add descriptions for the parameters?

File: pygaggle/data/segmentation.py (hunk 3 above)
Comment (justinborromeo) [reply pygaggle/data/segmentation.py:71]: Added one for the classes.  Is that good enough?

File: pygaggle/run/evaluate_trec_covid_ranker.py (hunk 4 above)
Comment (justinborromeo) [reply pygaggle/run/evaluate_trec_covid_ranker.py:117]: Just to be clear, do you mean setting the `help` string for each or just describing the parameters as comments in the code?

Comment (rodrigonogueira4) [reply pygaggle/run/evaluate_trec_covid_ranker.py:117]: for each parameter

File: pygaggle/data/segmentation.py (hunk 3 above)
Comment (rodrigonogueira4) [reply pygaggle/data/segmentation.py:71]: It is good to have a docstring for each method, even if it is only one or two sentences.

Comment (justinborromeo) [reply pygaggle/data/segmentation.py:71]: I added for the methods and classes.  I guess my question should have been whether I should add a docstring for the file?

Comment (rodrigonogueira4) [reply pygaggle/data/segmentation.py:71]: Sorry for the late reply. Ideally, there should be a docstring for the file, but I think it is fine docstrings for methods and classes only.

**Comment by rodrigonogueira4** [comment]
Hey @justinborromeo, let us know when this is ready for another review?

**Comment by justinborromeo** [comment]
Ready for review!

**Review by rodrigonogueira4 — APPROVED** [review]
Awesome, thanks for the hard work!

**Comment by rodrigonogueira4** [comment]
@justinborromeo do you have merge permissions?

**Comment by justinborromeo** [comment]
> @justinborromeo do you have merge permissions?

No :(

**Pull request merged**